bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/provider/ScriptModuleTypeProvider.java

```diff
@@ -145,26 +145,34 @@ public void removeProviderChangeListener(ProviderChangeListener<ModuleType> list
     public void setScriptEngineFactory(ScriptEngineFactory engineFactory) {
         ScriptEngine scriptEngine = engineFactory.createScriptEngine(engineFactory.getScriptTypes().get(0));
         if (scriptEngine != null) {
-            List<String> mimeTypes = new ArrayList<>();
-
             javax.script.ScriptEngineFactory factory = scriptEngine.getFactory();
-            String languageName = String.format("%s (%s)",
-                    factory.getLanguageName().substring(0, 1).toUpperCase() + factory.getLanguageName().substring(1),
-                    factory.getLanguageVersion());
-            mimeTypes.addAll(factory.getMimeTypes());
-            String preferredMimeType = mimeTypes.get(0);
-            mimeTypes.removeIf(mimeType -> !mimeType.contains("application") || mimeType.contains("x-"));
-            if (!mimeTypes.isEmpty()) {
-                preferredMimeType = mimeTypes.get(0);
-            }
-            parameterOptions.put(preferredMimeType, languageName);
+            parameterOptions.put(getPreferredMimeType(factory), getLanguageName(factory));
             logger.trace("ParameterOptions: {}", parameterOptions);
         } else {
             logger.trace("setScriptEngineFactory: engine was null");
         }
     }
 
-    public void unsetScriptEngineFactory(ScriptEngineFactory scriptEngineFactory) {
-        parameterOptions.clear();
+    public void unsetScriptEngineFactory(ScriptEngineFactory engineFactory) {
+        ScriptEngine scriptEngine = engineFactory.createScriptEngine(engineFactory.getScriptTypes().get(0));
```

Comment:
Not too nice to call "createScriptEngine" here, but it seems to be the only way to get hold of the mimeTypes to remove...

Reply:
Another way might have been to introduce another mapping to keep track of it but this also worked when I tested it.